v2/apiserver/internal/core/jobs.go

```diff
@@ -245,6 +245,10 @@ func (j *jobsService) Create(
 	eventID string,
 	job Job,
 ) error {
+	var useWorkspace = job.Spec.PrimaryContainer.WorkspaceMountPath != ""
+	var usePrivileged = job.Spec.PrimaryContainer.Privileged
+	var useDockerSocket = job.Spec.PrimaryContainer.UseHostDockerSocket
```

Comment:
I see you moved these lines up from farther down and that's fine, but there's a loop also that should have moved with it. It loops over the sidecar containers to determine if any of _them_ use these features.